### ISSUE TITLE
Switch to IndexedDB

### ISSUE THREAD

**Issue by aidarkhanov** [samuelgozi/firebase-auth-lite#26]
Hello, thank you for this project! I want to leave an issue to let you know, that I am working on the project to using IndexedDB instead of local storage. Here are the reasons:

1) By leveraging IndexedDB, one is able to use firebase-auth-lite in Service Worker setting;
2) Browser extensions work better with IndexedDB.

The benefits above are also [communicated](https://groups.google.com/d/msg/firebase-talk/wgSvjniKPQI/5BUymV7gAgAJ) by the Firebase team.

Also, what do you think of making IndexedDB default and only storage?

**Comment by samuelgozi** [comment]
@aidarkhanov Thank you for submitting this pull request.
Initially, I decided to go with local storage because of the better browser support([specifically opera mini which is widely used in Africa](https://chenhuijing.com/blog/we-need-to-talk-about-opera-mini/#🇲🇾)).

However, you made some good points. Since I do intend on using service workers in the near future with this SDK it would make sense to move to IndexedDB, the only issue I see there is that the API is very difficult to work with, and using a library like IDB will double the size of this lib.

It is currently very easy to extend this library to work with other storage solutions. The constructor of Auth receives a `storage` property that should be an object with the signature:
```ts
interface Storage {
	/** Save to local storage */
	set(key: string, value: string): Promise<void>;
	/** Get from local storage */
	get(key: string): Promise<string | null>;
	/** Remove from local storage */
	remove(key: string): Promise<void>;
}
```

And the current implementation is close to:
```ts
const storageApi: Storage = {
	async set(k: string, v: string) {
		return localStorage.setItem(k, v);
	},

	async get(k: string) {
		return localStorage.getItem(k);
	},

	async remove(k: string) {
		return localStorage.removeItem(k);
	}
};
```

So you can use this as a reference for IndexedDB.

To recap; I will switch to IndexedDB in the future(probably before v1.0.0) but I can't yet tell you exactly when since I'm currently focused on refactoring `firebase-firestore-lite`, only after I will switch my focus to this lib.

Please let me know if you have any suggestions. I will leave this issue open for discussion and untill it is implemented(or otherwise).

**Comment by andymatuschak** [comment]
> To recap; I will switch to IndexedDB in the future

FWIW, there are some contexts where local storage will remain the better option: for example, Safari forbids IDB access in iframes. I like your dependency injection approach!

**Comment by samuelgozi** [comment]
@andymatuschak thank, I didn’t know that. I will take that into consideration.
